File: Projects/leetcode/maxSubArray.kt
package leetcode

import kotlin.math.max

fun maxSubArray(nums: IntArray): Int {
    var sum: Int = 0
    var maxSum: Int = Int.MIN_VALUE
    for (num in nums) {
        sum = if (sum >= 0) sum + num else num
        maxSum = max(sum, maxSum)
    }
    return maxSum
}

File: 專案/leetcode/max_sub_array.rs
use std::cmp::max;

pub fn max_sub_array(nums: Vec<i32>) -> i32 {
    let mut cur_max_sum = nums[0];
    let mut max_sum = nums[0];
    for &num in &nums[1..] {
        cur_max_sum = max(num, cur_max_sum + num);
        max_sum = max(max_sum, cur_max_sum);
    }
    return max_sum;
}
